### PR TITLE
Add previously missing cursor types to JDBC utils.

### PR DESCRIPTION
**Pull request by davinchia**
## What
Incremental syncs using NVARCHAR as the key was previously failing since we did not recognise the type.

This was reported by a user.

Take the chance to fix wrong casting of JDBC integers to JSON Schema Numbers type, that is resulting in the data being transformed as a float after normalisation.

## How
Add the missing types in. 

Bump all connector versions that use the class:
- source-mssql
- source-mysql
- source-redshift
- source-postgres
- destination-postgres
- desintation-redshift
- destination-snowflake

## Pre-merge Checklist
- [ ] *Run integration tests*
- [ ] *Publish Docker images*

## Recommended reading order
1. JdbcUtils.java

### CODE REVIEW

**Comment by ChristopheDuong** [comment]
Oh great, since you are tackling this, could you maybe add support for `integer` as well?

Thanks!

https://github.com/airbytehq/airbyte/issues/1006#issuecomment-794386288

File: airbyte-config/init/src/main/resources/config/STANDARD_SOURCE_DEFINITION/decd338e-5647-4c0b-adf4-da0e75f5a750.json
Comment (ChristopheDuong) [inline airbyte-config/init/src/main/resources/config/STANDARD_SOURCE_DEFINITION/decd338e-5647-4c0b-adf4-da0e75f5a750.json:5]: I think there's something we don't handle very well at the moment is conflict in connector's versions being bumped in multiple branches at the same time... I'm not sure how we should resolve this though.

So when you'll try to bump versions of those connectors, it's going to fail because 0.2.2 has already been published but not merged yet...:

https://hub.docker.com/r/airbyte/source-postgres/tags?page=1&ordering=name

Comment (davinchia) [reply airbyte-config/init/src/main/resources/config/STANDARD_SOURCE_DEFINITION/decd338e-5647-4c0b-adf4-da0e75f5a750.json:5]: that makes sense. will coordinate with artem!

**Review by ChristopheDuong — APPROVED** [review]

File: airbyte-db/src/test/java/io/airbyte/db/jdbc/TestJdbcUtils.java
Comment (jrhizor) [inline airbyte-db/src/test/java/io/airbyte/db/jdbc/TestJdbcUtils.java:283]: Does this mean we expect destinations to handle all `INTEGER` as longs?

File: airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java
Comment (jrhizor) [inline airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java:201]: Does something need to change in normalization to support the new primitive type `INTEGER`?

Comment (ChristopheDuong) [reply airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java:201]: No, other sources were already producing `integer` columns as json primitives

File: airbyte-db/src/test/java/io/airbyte/db/jdbc/TestJdbcUtils.java
Comment (cgardens) [reply airbyte-db/src/test/java/io/airbyte/db/jdbc/TestJdbcUtils.java:283]: 1. Do we need to add integer in this PR?
2. How good to we feel that everything still works after adding a new type? Just running the tests is insufficient in this case because none of the tests are testing for integer type. So it seems like we need to add tests for sources, destinations, etc to make sure that this gets supported sanely. We can do this, but that's a bigger PR (which goes back to question 1). 

My feeling is as is, this is yoloing a kinda big change. So we should know why it is valuable right now and understand why we are yoloing it versus doing it more carefully with adequate testing.

File: airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java
Comment (cgardens) [reply airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java:201]: i think more broadly the question is just what needs to change downstream to support the integer type?

Comment (cgardens) [reply airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java:201]: this may be a little puritanical of me, but it seems a little odd to have a comment about normalization in this utils class. feel free to keep it if you feel it's helpful though. (2 of 10 on the scale)

**Review by cgardens — CHANGES_REQUESTED** [review]
the nvarchar parts look good to me. see comments below that I am wary about doing the integer piece. curious to understand if we need to do that now.

File: airbyte-db/src/test/java/io/airbyte/db/jdbc/TestJdbcUtils.java
Comment (ChristopheDuong) [reply airbyte-db/src/test/java/io/airbyte/db/jdbc/TestJdbcUtils.java:283]: normalization is converting a `JsonSchemaPrimitive.INTEGER` into `dbt_utils.type_bigint` (the largest integer available as we don't have info on the size of the integer anymore)

```
{% macro default__type_bigint() %}
    bigint
{% endmacro %}

{% macro bigquery__type_bigint() %}
    int64
{% endmacro %}
```
from https://github.com/fishtown-analytics/dbt-utils/blob/ceb28497769c642cae7e3d5d18f1fe6bb253ef59/macros/cross_db_utils/datatypes.sql#L71

File: airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java
Comment (ChristopheDuong) [reply airbyte-db/src/main/java/io/airbyte/db/jdbc/JdbcUtils.java:201]: In normalization, it is already handling `integer` fields.

For example facebook catalog produces such as this one: https://github.com/airbytehq/airbyte/blob/f5094b558ad77a5985a54bd76c2cfca9a891a027/airbyte-integrations/bases/base-normalization/unit_tests/resources/facebook_catalog.json#L104

**Review by cgardens — APPROVED** [review]
Talked offline. Agreed to separate out the integer stuff into a separate project. Approving because once that's done this can be merged. Don't want to block due to time zone.

**Comment by davinchia** [comment]
@Chris I'm going to separate this out for now to make this change cleaner. Let's talk later on how to approach this from a test perspective. Looks like we might have other types to add as well.

**Pull request merged**
